docs/DeveloperGuide.md

```diff
@@ -1,34 +1,978 @@
 # Developer Guide
 
-## Design & implementation
+## Table of Contents
+- [Introduction](#10-introduction)
+- [Setting up](#20-setting-up)
+- [Design](#30-design)
+  * [Architecture](#31-architecture)
+  * [UI component](#32-ui-component)
+  * [Logic component](#33-logic-component)
+  * [Model component](#34-model-component)
+  * [Storage component](#35-storage-component)
+- [Implementation](#40-implementation)
+  * [Book Management](#41-feature-book-management)
+  * [Quote Management](#42-feature-quote-management)
+  * [Progress Tracker](#43-progress-tracker)
+  * [Category Management](#44-feature-category-management)
+  * [Rating System for books](#45-feature-rating-system-for-books)
+- [Appendix: Requirements](#appendix-requirements)
+  * [Product Scope](#product-scope)
+  * [User Stories](#user-stories)
+  * [Non-Functional Requirments](#non-functional-requirements)
+  * [Glossary](#glossary)
+- [Appendix: Instructions for manual testing](#appendix-instructions-for-manual-testing)
```

Comment:
Should this part be divided into different segments? For new developers, it's hard to catch up with the whole bunch of commands.
![appendix](https://user-images.githubusercontent.com/55688719/97389033-689b9880-1914-11eb-8952-82773a03f194.PNG)

Reply:
I agree with you. Additionally, could you try to partition each segment based on functionality?